text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,601 @@
+    1. Kubectl - This one has two dependencies to keep
+       updated. All updates described below will result in a minor version bump
+       to this package.
+        1. [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+            1. Kubernetes users will need to be able to choose the minor version
+               of kubectl that they want to use. And the aws-eks module will
+               need to as well.
+            2. We will release multiple packages, one for each currently
+               supported minor version of kubernetes, e.g.
+               `@aws-cdk/lambda-layer-kubectl-v20`. Each one will expose a
+               Construct called `KubectlLayer`.
+            3. A weekly automated task will check the [Kubernetes
+               API](https://dl.k8s.io/release/stable.txt) for a new patch
+               version of each currently supported minor version, and
+               automatically update the version of kubectl used in each package.
+               This update will be performed with an auto-approved PR.
+            4. A weekly automated task will check for new supported minor
+               versions of kubectl. If there is one, a PR will be automatically
+               be created which creates a new
+               `@aws-cdk/lambda-layer-kubectl-XYZ` package for the new minor
+               version. A human, the on-call engineer, will need to review this
+               one.
+        2. [helm](https://helm.sh/) is also included in each asset in
+           these packages
+            1. Helm has a somewhat complicated version support policy documented
+               [here](https://helm.sh/docs/topics/version_skew/).
+            2. For each `@aws-cdk/lambda-layer-kubectl-XYZ` package, we will
+               include the latest available version of Helm which is compatible
+               with that version of Kubernetes.
+            3. A weekly automated task will update the minor and patch versions
+               of helm within the supported ranges.
+            4. The automated task that checks for new supported minor versions
+               of kubectl, will also attempt to figure out what is the latest
+               version of helm compatible with that version of kubectl. This
+               will be verified by a human in PR review to make sure it is
+               correct.
+    2. AWS CLI - We will release two separate packages to make major versions 1
+       and 2 both available, and automatically update to the latest minor or
+       patch version.
+        1. Two packages: `@aws-cdk/lambda-layer-aws-cli-v1`, and
+           `@aws-cdk/lambda-layer-aws-cli-v2`.
+        2. Automatically check for new minor and patch versions from PyPi -
+           [awscli](https://pypi.org/project/awscli/),
+           [awscliv2](https://pypi.org/project/awscliv2/). Each update will
+           result in a minor version bump of the corresponding package.
```

Comment:
should i change this to reflect the actual awscliv2 implementation?

Reply:
Yes! It would be great if you could add that!